pkg/cache/cache.go

```diff
@@ -90,8 +91,22 @@ type Informer interface {
 type ObjectSelector internal.Selector
 
 // SelectorsByObject associate a client.Object's GVK to a field/label selector.
+// Use SelectorsByObjectDefaultKey as key to specify a default selector that
+// will be used for all types that do not have a selector configured.
 type SelectorsByObject map[client.Object]ObjectSelector
 
+// SelectorsByObjectDefaultKey can be used in SelectorsByObject to configure a default
+// selector for all kinds that do not have a specific selector set up.
+type SelectorsByObjectDefaultKey struct{ metav1.Object }
+
+// GetObjectKind implements runtime.Object.
+func (s *SelectorsByObjectDefaultKey) GetObjectKind() schema.ObjectKind {
+	return schema.EmptyObjectKind
+}
+
+// DeepCopyObject implements runtime.Object.
+func (s *SelectorsByObjectDefaultKey) DeepCopyObject() runtime.Object { return s }
```

Comment:
Rather than encoding the default selector into the `SelectorsByObject` option which requires this new exported `SelectorsByObjectDefaultKey` type, is it possible to add a `DefaultSelector` field to the options struct and then handle the setup internally?

Reply:
That's a great idea, updated the PR, PTAL